src/api/gateway.go

```diff
@@ -19,14 +20,16 @@ type Gateway struct {
 	*daemon.Daemon
 	*visor.Visor
 	*wallet.Service
+	*note.Manager
 }
 
 // NewGateway creates a Gateway
-func NewGateway(d *daemon.Daemon, v *visor.Visor, w *wallet.Service) *Gateway {
+func NewGateway(d *daemon.Daemon, v *visor.Visor, w *wallet.Service, n *note.Manager) *Gateway {
```

Comment:
I'm not sure about adding a new parameter here . Could this be implemented by keeping signature as it used to be and adding a new `GetNotesManager()` pointer receiver (or alike) ?

Reply:
It could be. But it wouldn't look similar to other dependencies then.

Reply:
@olemis the extra parameter here is the correct way